rust/openvasd/src/controller/entry.rs

```diff
@@ -134,165 +107,128 @@ where
-                                            "Unable to start Scan with ID {}",
-                                            progress
-                                                .scan
-                                                .scan_id
-                                                .as_ref()
-                                                .unwrap_or(&"".to_string())
-                                        );
-                                        match e {
-                                            Error::SocketDoesNotExist(path) => {
-                                                Ok(ctx.response.service_unavailable(
-                                                    "OSP",
-                                                    format!(
-                                                        "socket {} does not exist",
-                                                        path.display()
-                                                    )
-                                                    .as_str(),
-                                                ))
-                                            }
-                                            _ => Ok(ctx.response.internal_server_error(&e)),
-                                        }
-                                    }
-                                }
-                            }
-                            (models::Action::Stop, Some(progress)) => {
-                                tracing::debug!(
-                                    "{}: {}",
-                                    progress.scan.scan_id.as_ref().unwrap_or(&"".to_string()),
-                                    action.action,
-                                );
-                                match ctx.scanner.stop_scan(progress) {
-                                    Ok(()) => Ok(ctx.response.no_content()),
-                                    // TODO we need to parse the ospd response for status code
+            match crate::request::json_request::<models::ScanAction>(&ctx.response, req)
+                .await
+                .map(|a| a.action)
+            {
+                Ok(models::Action::Start) => {
+                    let (scan, mut status) = ctx.db.get_decrypted_scan(&id).await?;
+                    if status.is_running() {
+                        use models::Phase::*;
+                        let expected = &[Stored, Stopped, Failed, Succeeded];
+                        Ok(ctx.response.not_accepted(&status.status, expected))
+                    } else {
+                        match ctx.scanner.start_scan(scan).await {
+                            Ok(_) => {
+                                status.status = models::Phase::Requested;
+                                match ctx.db.update_status(&id, status).await {
+                                    Ok(_) => Ok(ctx.response.no_content()),
                                     Err(e) => Ok(ctx.response.internal_server_error(&e)),
```

Comment:
I think we also have to stop the scan at this point, as
```rust
ctx.scanner.start_scan(scan).await
```
was successfully executed